### ISSUE TITLE
"Please share your thoughts on the matter ..." as the notice requests.

### ISSUE THREAD

**Issue by sdrausty** [termux/termux-packages#1930]
This article is being considered for deletion in accordance with Wikipedia's deletion policy. Please share your thoughts on the matter at this article's entry on the Articles for deletion page. https://en.m.wikipedia.org/wiki/Termux
https://en.m.wikipedia.org/wiki/Termux#/issues

@fornwall why is news of this deleted from Termux G+ https://plus.google.com/109614420080028230010/posts/Nj6RLZdirRu This topic was first posted six days ago, only to be deleted repeatedly.  Without group support, this page https://en.m.wikipedia.org/wiki/Termux along with these https://en.m.wikipedia.org/w/index.php?search=Termux&title=Special:Search&profile=default&fulltext=1&searchToken=32xpund017jy7v2bdp7cnnrhu six accompanying pages will be deleted as well. **If Termux users don't know of this, they will never "Please share your thoughts on the matter ..." as the notice requests.**

Who deleted https://plus.google.com/109614420080028230010/posts/iX7yD7rg3F1 a request for this issue resolution six days ago?  Whoever deleted this post and is deleting similar  postings is keeping  Termux from developing into a mature hybrid Linux distribution.

![screenshot_20171212-111835](https://user-images.githubusercontent.com/27742457/33895306-3aeb46b0-df2e-11e7-9dcd-22c7f0f03fa4.png)

**Comment by sdrausty** [comment]
To help save the Termux Wikipedia article from deletion, add **Do not delete** + (reasons why) to https://en.m.wikipedia.org/wiki/Wikipedia:Articles_for_deletion/Termux#/editor/1 this page at the bottom of the page.

**Comment by sdrausty** [comment]
**Wikipedia is the 5th most popular website on the Internet with an estimated one billion page views per day. ** https://www.techspot.com/news/71800-study-77-wikipedia-articles-written-1-editors.html

**Comment by sdrausty** [comment]
__Wikipedia is the 5th most popular website on the Internet with an estimated one billion page views per day.__  https://www.techspot.com/news/71800-study-77-wikipedia-articles-written-1-editors.html

To help save the Termux Wikipedia articles from deletion, add __Do not delete + (reasons why)__ to https://en.m.wikipedia.org/wiki/Wikipedia:Articles_for_deletion/Termux#/editor/1 this page at the bottom of the page.

**Comment by sdrausty** [comment]
A simpler solution may be adding meaningful sections and edits to https://en.m.wikipedia.org/wiki/Wikipedia:Articles_for_deletion/Termux page in order to close this issue asap.

**Comment by sdrausty** [comment]
Latest update https://en.m.wikipedia.org/wiki/Wikipedia:Articles_for_deletion/Termux Wikipedia page deleted.

**Comment by sdrausty** [comment]
Cut and paste this page back online if you want to have Termux well known with many devoted developers.

{{Infobox software
| name = Termux
| logo = 
| logo alt =
| logo caption =
| screenshot = 
| screenshot alt =
| caption =
| collapsible = 
|  author = [https://github.com/fornwall Fredrik Fornwall]
| developer = [https://github.com/termux Termux at GitHub]
| released = 
| discontinued = 
| ver layout = 
| latest release version = [https://github.com/termux/termux-app/releases v0.59]
| latest release date = 
| latest preview version =
| latest preview date = 
| repo = [https://github.com/termux?tab=repositories Termux Repositories]
| status =
| programming language = [[Java (programming language)|Java]]
| operating system = [[Android (operating system)|Android]], [[Chrome_OS |Chromebook]] and [[Fire_OS |Fire OS]]
| platform =
| size =
| language =
| language count = 
| language footnote =
| genre = [[Command-line_interface |Command-line interface]]
and [[Terminal_emulator |Terminal emulator]]
| license = [[Open-source_model |Open source]]
| alexa =
| website = [https://termux.com Termux] and [https://wiki.termux.com The Termux Wiki]
| standard =
| AsOf =
}}

'''Termux''' is an [[open source]] [[terminal emulator]] and [[Linux]] [[Computer_architecture |environment]] app for [[Android (operating system)|Android]],<ref>{{cite web|url=https://www.xda-developers.com/termux-the-ultimate-linux-terminal-emulator-for-android-xda-spotlight/ |title=Termux the ultimate linux-terminal emulator for android xda spotlight |publisher=xda-developers}}</ref><ref>{{cite web|url=http://cecs.wright.edu/~pmateti/Courses/4440/Lectures/Internals/Android-Internals-1.html |title=Android OS Internals |publisher=Wright State University}}</ref><ref>{{cite web|url=http://cereal.forest.usf.edu/alessio/technology.php |title=Technology |publisher=University of South Florida}}</ref> [[ChromeOS |Chromebook]],<ref>{{cite web|url=https://medium.freecodecamp.org/how-i-built-a-web-server-using-go-and-on-chromeos-3b83e4c2da5f |title=How I built a web server using Go on ChromeOS |publisher=Peter Gleeson}}</ref><ref>{{cite web|url=https://blog.lessonslearned.org/building-a-more-secure-development-chromebook/ |title=My $169 development Chromebook |publisher=Kenneth White}}</ref> and [[FireOS |Fire OS]].<ref>{{cite web|url=https://www.howtogeek.com/232726/how-to-install-the-google-play-store-on-your-amazon-fire-tablet/ |title=How to Install the Google Play Store on the Amazon Fire Tablet or Fire HD 8 |publisher=How-To Geek}}</ref>  '''It works with no rooting and setup required.'''<ref>{{cite web|url=https://termux.com |title=Termux |publisher=Termux}}</ref>  Running full Linux distributions with and without an [[X_Window_System |X Window System]] on a smartphone and tablet want to be configured before activation.  Advanced features such as typesetting on a smartphone through employing [[TeX_Live |TeX Live]] and similar are available as Termux packages.  Use <code>pkg install <package></code> to install packages in Termux on your smartphones, tablets and similar devices online today.


A minimal [[List_of_GNU_packages#Base_system |base system]] is installed automatically in Termux; Additional packages are available using the [[APT_(Debian) |APT]] package manager.
With Termux you get a full [[Linux]] experience with [[Bash_(Unix_shell) |BASH]],<ref>{{cite web|url=https://termux.com/common-packages.html |title=Common Packages |publisher=Termux}}</ref> [[Friendly_interactive_shell |FISH]], [[Mosh_(software) |MOSH]], [[Tcsh |TCSH]], [[Z_shell |ZSH]] and even things like [[Clang]],<ref>{{cite web|url=https://www.marcbilodeau.com/termux/ |title=Termux, the best linux terminal for Android |publisher=Marc Bilodeau}}</ref> [[Eclipse_Compiler_for_Java |ECJ]], [[Emacs]],<ref>{{cite web|url=https://emacs.stackexchange.com/questions/37301/how-to-build-emacs-from-sources-on-termux |title=How to build Emacs from sources on Termux |publisher=Emacs Stack Exchange}}</ref> [[GNU_Compiler_Collection |GCC]], [[Gradle]],
[[Javac]], [[GNU_nano |Nano]],<ref>{{cite web|url=https://wiki.termux.com/wiki/Text_Editors |title=Text Editors |publisher=The Termux Wiki}}</ref>  [[Vim_(text_editor)#Neovim |Neovim]]<ref>{{cite web|url=https://github.com/neovim/neovim/wiki/Installing-Neovim |title=Installing Neovim |publisher=Neovim}}</ref> and [[Vim_(text_editor) |Vim]].<ref>{{cite web|url=https://wiki.termux.com/wiki/Vimtutor |title=Vimtutor |publisher=The Termux Wiki}}</ref>  Programming languages such as [[Assembly_language |Assembly]], [[C_(programming_language) |C]], [[The_C%2B%2B_Programming_Language |C++]], [[Haskell_(programming_language) |Haskel]], [[Java_(programming_language) |Java]], [[Lisp]], [[Node.js]],<ref>{{cite web|url=https://medium.freecodecamp.org/building-a-node-js-application-on-android-part-1-termux-vim-and-node-js-dfa90c28958f |title=Building a Node.js application on Android - Part 1: Termux, Vim and Node.js |publisher=Medium}}</ref> [[Octave_(programming_language) |Octave]], [[OCaml]],<ref>{{cite web|url=http://ygrek.org.ua/p/ocaml-termux.html |title=OCaml Native on Android! |publisher=ygrek}}</ref> [[PHP]],<ref>{{cite web|url=https://www.phpclasses.org/blog/package/10192/post/1-Tutorial-on-How-to-Control-an-Android-Phone-or-Tablet-using-PHP-with-Termux-App.html |title=How to Use Termux Tutorial Running PHP Web Server or CLI with Termux PHP App to Control an Android Phone or Tablet - PHP Termux API package blog |publisher=PHPClasses}}</ref> [[Perl]], [[Picolisp]],<ref>{{cite web|url=https://picolisp.com/wiki/?PilBox |title=PilBox - Building Android Apps in PicoLisp |publisher=PicoLisp}}</ref> [[Python_(programming_language) |Python]],<ref>{{cite web|url=https://pypi.python.org/pypi/termux-create-package/0.4 |title=termux-create-package 0.4 |publisher=Python Software Foundation}}</ref> [[R_(programming_language) |R]],<ref>{{cite web|url=https://wiki.archlinux.org/index.php/R |title=R is a "free software environment for statistical computing and graphics." |publisher=ArchWiki}}</ref> [[Ruby_(programming_language) |Ruby]],<ref>{{cite web|url=http://www.linux-magazine.com/Online/Blogs/Productivity-Sauce/Termux-Lightweight-Linux-Environment-for-Android |title=Termux: Lightweight Linux Environment for Android |publisher=Linux New Media USA, LLC}}</ref> [[Rust_(programming_language) |Rust]]<ref>{{cite web|url=https://k33g.github.io/2017/08/05/TERMUX.html  |title=OnBoard programming on Android with Termux |publisher=@k33g_org}}</ref> and [[TeX_Live |TeX]]<ref>{{cite web|url=https://wiki.termux.com/wiki/TeX_Live  |title=TeX Live Package Management |publisher=The Termux Wiki}}</ref> are included.  And it's just a normal [[Android_application_package |app]] that doesn't require [[Rooting_(Android) |rooting]] and similar.


Once installed Linux distributions can be installed in a [[chroot]] container with [https://wiki.termux.com/wiki/PRoot Termux PRoot]<ref>{{cite web|url=https://wiki.termux.com/wiki/PRoot  |title=PRoot |publisher=The Termux Wiki}}</ref> in Termux on a smartphone.  These Linux distributions are known to work in Termux: [[Arch_Linux |Arch]],<ref>{{cite web|url=https://sdrausty.github.io/TermuxArch/ |title=TermuxArch |publisher=@sdrausty}}</ref>
[[ArchStrike_Linux |ArchStrike]],<ref>{{cite web|url=https://sdrausty.github.io/TermuxArch/docs/install |title=Install TermuxArch |publisher=@sdrausty}}</ref>
[[BlackArch_Linux |BlackArch]],<ref>{{cite web|url=https://sdrausty.github.io/TermuxArch/docs/install |title=Install TermuxArch |publisher=@sdrausty}}</ref>
[[Debian]],<ref>{{cite web|url=https://github.com/sp4rkie/debian-on-termux |title=debian-on-termux |publisher=@sp4rkie}}</ref>
[[Fedora_(operating_system) |Fedora]],<ref>{{cite web|url=https://nmilosev.svbtle.com/termuxfedora-install-fedora-on-your-phone-with-termux |title=Termux-Fedora - install Fedora on your phone with Termux |publisher=@nmilosev}}</ref>
[[Kali_Linux_NetHunter_Edition |Kali Nethunter]],<ref>{{cite web|url=https://github.com/Hax4us/Nethunter-In-Termux |title=Nethunter-In-Termux |publisher=@Hax4us}}</ref>
[[Slackware]]<ref>{{cite web|url=https://github.com/gwenhael-le-moine/TermuxSlack |title=TermuxSlack |publisher=@gwenhael-le-moine}}</ref> and
[[Ubuntu (operating system)|Ubuntu]].<ref>{{cite web|url=https://github.com/Neo-Oli/termux-ubuntu |title=termux-ubuntu |publisher=@Neo-Oli}}</ref>  Termux is being actively developed.  Hopefully other [[List_of_Linux_distributions |Linux distributions]] are also being tested, and are being prepared for publication.  Take part in Termux on smartphone, tablet and similar devices.  Consider joining our Termux [https://wiki.termux.com/wiki/Community Community.]  You are welcome at [https://github.com/termux/ Termux!]

==Addons==

Termux has some extra features. You can add them by installing addons:

'''[https://wiki.termux.com/wiki/Termux:API Termux:API]'''<ref>{{cite web|url=http://www.linux-magazine.com/Issues/2016/193/Workspace-Termux |title=Tiny Linux |publisher=Linux New Media USA, LLC}}</ref>
Access Android, Chromebook and Fire OS hardware features.

'''[https://wiki.termux.com/wiki/Termux:Boot Termux:Boot]'''<ref>{{cite web|url=https://www.ostechnix.com/termux-run-linux-android-devices-no-rooting-required/ |title=Termux – Run Linux On Android Devices, No Rooting Required! |publisher=OSTechNix}}</ref>
Run script(s) when your device boots.

'''[https://wiki.termux.com/wiki/Termux:Float Termux:Float]'''<ref>{{cite web|url=https://hackernoon.com/code-a-react-website-on-your-android-smartphone-or-tablet-like-you-never-imagined-3e56c534f6e7 |title=Code a React website on your Android smartphone or tablet like you never imagined |publisher=Medium}}</ref>
Run Termux in a floating window.

'''[https://wiki.termux.com/wiki/Termux:Styling Termux:Styling]'''<ref>{{cite web|url=https://www.gammerson.com/2017/03/termux-ultimate-linux-terminal-emulator-android.html |title=Termux – The Ultimate Linux Terminal Emulator for Android |publisher=Gammerson}}</ref>
Have color schemes and powerline-ready fonts customize the appearance of the Termux terminal.

'''[https://wiki.termux.com/wiki/Termux:Task Termux:Task]'''<ref>{{cite web|url=http://downloads.zdnet.com/product/20432-77503847/ |title=Termux:Task |publisher=ZDNet CBS Interactive}}</ref>
An easy way to call Termux executables from Tasker and compatible apps.

'''[https://wiki.termux.com/wiki/Termux:Widget Termux:Widget]'''<ref>{{cite web|url=http://downloads.zdnet.com/product/2072-77367075/ |title=Termux:Widget |publisher=ZDNet CBS Interactive}}</ref>
Start small scriptlets from the home screen.


==References==
{{reflist}}

==External links==

* [https://wiki.termux.com/wiki/Community Community] at [https://wiki.termux.com/wiki/Community The Termux Wiki]
* [https://wiki.termux.com/wiki/FAQ FAQ] at [https://wiki.termux.com/wiki/FAQ The Termux Wiki]
* [[IRC]]: Channel #Termux on [https://webchat.freenode.net/ freenode] that is mirrored with [https://gitter.im/termux/termux Gitter].
* [https://github.com/termux/ Github]: Source code and project repository at GitHub.  Submit bug reports regarding the main app, packages, and addons. Create pull requests for your solutions.
* [https://gitter.im/termux/termux Gitter]: Chatroom that is mirrored with [[IRC]].
* [http://termux.com Termux]

[[Category:Terminal emulator software that uses GTK+]]
[[Category:Android (operating system)]]
[[Category:Chrome (operating system)]]
[[Category:Fire (operating system)]]
{{software-stub}}

**Comment by sdrausty** [comment]
The simplest response to Wikipedia's exclusion of Termux is worldwide class action lawsuit.  Do NOT include terrorists, include scientific us instead.  Do not proffess science while promoting terrorists and similar.

**Comment by its-pointless** [comment]
Calm down. Wikipedia is just being wikipedia. The more you fight the more they justify doing bullshit. Its been a clusterfuck for years. There is only pain in fighting wikipedia.

**Comment by RolandR** [comment]
Hey, @sdrausty.

I took a look at the article you wrote, and I have to say, I'm not surprised that it was deleted.  
There are several issues with it, most severely:
 * It reads like an ad. Things like this should be avoided at all cost:
>Take part in Termux on smartphone, tablet and similar devices. Consider joining our Termux Community. You are welcome at Termux!
* Most of the article is just long lists of programming languages or packages or somesuch. That makes it pretty tedious to read, and I don't think somebody who's never heard of Termux before would care.
* The article, and its references, should make it obvious how and why Termux is of encyclopedic relevance.
* Scrap the list of Addons, or incorporate it into the article text.
* Wikipedia is not a tutorial. It should describe what Termux *is*, not *how to use it*.

I suggest you take some time to have a look at [Contributing to Wikipedia](https://en.wikipedia.org/wiki/Wikipedia:Contributing_to_Wikipedia) and browse the pages linked there. Also useful are [What Wikipedia is not](https://en.wikipedia.org/wiki/Wikipedia:What_Wikipedia_is_not), [Writing better articles](https://en.wikipedia.org/wiki/Wikipedia:Writing_better_articles) and [Relevance of content](https://en.wikipedia.org/wiki/Wikipedia:Relevance_of_content).

Also, as @its-pointless wrote: Calm down. Not everything has to be in Wikipedia.  
Minecraft was at some point deemed irrelevant, too. If Termux isn't there yet, just relax and give it time.

**Comment by sdrausty** [comment]
It appears that this https://plus.google.com/+ScottMortimer moderator was blocking news from our community members about the now deleted Termux Wikipedia article at Termux G+ while it was online for more than a week.

**Comment by sdrausty** [comment]
@RolandR @its-pointless thanks for your interest and insight.  The last edits to the Termux Wikipedia page were fulfilling your suggestions of toning down the marketing in the page as you suggested @RolandR.  That's when it went offline.   @vishslbiswas and other gitter.im/termux/termux users had critiqued the page a day before.  As far as "long lists of programming languages" and similar:
1) These are very important keywords that matter for search results once indexed by Internet web crawlers.
2) Developers use this terminology regularly.  It is important to us when we want A+B results.  
3) It is a fantastic read simply because it enumerates and crosslinks what is possible in Termux.  A long list of packages and similar crosslinked at Wikipedia answers an often asked Termux question, "What can I do with Termux?"

**Comment by sdrausty** [comment]
Our **encyclopedia tabloid** still refuses to publish about https://en.m.wikipedia.org/wiki/Special:Search?search=Termux us by deleting references to Termux in other packages/systems run in Termux on device. 
> TeX Live 
for several Linux distributions such as openSUSE, Fedora, Debian, Ubuntu, Termux and Gentoo. Other Unix operating systems like OpenBSD, FreeBSD and NetBSD
10 KB (429 words) - 00:18, 9 December 2017

There were many pages that *had* links to https://en.m.wikipedia.org/wiki/Termux in December 2017.  At present this page displays zero/null information and all the referring pages, sanse one, have been stripped of links to https://en.m.wikipedia.org/wiki/Termux with the referring information. 

Define:**tabloid** NORTH AMERICAN
sensational in a lurid or vulgar way

**Comment by Quasic** [comment]
Generally, each page can have its own consensus, following sources on that particular subject. Many pages also follow consensus for particular page groups, similar to classes. One problem with wikis is finding where links have been removed. If you have a list of link removals, we could look at them. The main things to keep in mind on Wikipedia are cooperation and understanding. These people can be friends. Also, everything changes. Try to not worry about it. It doesn't usually matter so much in the long term.

**Comment by sdrausty** [comment]
@Quasic Termux was recently deleted from these articles:  
https://en.wikipedia.org/wiki/Comparison_of_Linux_distributions
https://en.wikipedia.org/wiki/List_of_Linux_distributions
How would you incorporate Termux into them?

> If you have a list of link removals, we could look at them. 

This should be fairly simple to recreate.  Follow this one surviving link as an example: 
https://en.wikipedia.org/wiki/TeX_Live
> TeX Live is a free software distribution for the TeX typesetting system that includes major TeX-related programs, macro packages, and fonts. It is the replacement of its no-longer supported[1] counterpart teTeX.[2] It is now the default TeX distribution for several Linux distributions such as openSUSE,[3] Fedora,[4] Debian,[5][6] Ubuntu,[7] Termux[8] and Gentoo.[9] Other Unix operating systems like OpenBSD,[10] FreeBSD[11] and NetBSD[12] have also converted from teTeX to TeX Live.

Search for similar Termux packages listed on Wikipedia and add relevant Termux information.

**Comment by Quasic** [comment]
I didn't find termux in the comparison article's history.

I found https://en.wikipedia.org/w/index.php?title=List_of_Linux_distributions&diff=prev&oldid=818865042 but list articles usually require items to currently have their own articles. There will be no problem relisting termux here once it has an article.

Using wikis as references is a bit controversial. In the last case you posted, a different reference would probably be wanted, but it could be ok for now.

It looks like other pages I can look at later are listed at https://en.wikipedia.org/wiki/Special:Contributions/SDRausty

There is a big learning curve there. People often understand mistakes, which are common. The closer's talk page is the appropriate page for getting a deleted page restored. Many get that wrong. I noticed your edits to the closer's talk page were in the wrong place, though. It is easy to post near the top instead of using the new section button, but I know I find it very annoying on my talk page. An apology for this could help, just make sure to use the new section button (or new topic or whatever it may be called.) You could ask for the article to be sandboxed or drafted, but it is pointless to ask about the deletion or restoring the article as it was. The closer really had little choice in the matter, and is responsible to many people to not bend too much.

**Comment by sdrausty** [comment]
@Quasic https://github.com/termux/termux-packages/issues/1930#issuecomment-356814825 and https://wiki.termux.com/index.php?title=WikipediaEN&diff=3648&oldid=3554 have the article's state prior to deletion and locking up at Wikipedia.  

> Revision as of 00:31, 9 December 2017

Template:Article for deletion/dated Template:Refimprove Template:Infobox software 

Termux is an open source terminal emulator and Linux environment app for Android<ref>Template:Cite web</ref><ref>Template:Cite web</ref><ref>Template:Cite web</ref>, Chromebook<ref>Template:Cite web</ref><ref>Template:Cite web</ref> and Fire OS. It works directly with no rooting or setup required. A minimal base system is installed automatically; Additional packages are available using the APT package manager. 

If you install Termux, you get a full Linux experience with BASH<ref>Template:Cite web</ref>, FISH, MOSH, TCSH, ZSH, and even things like Clang<ref>Template:Cite web</ref>, Emacs<ref>Template:Cite web</ref>, GCC, Nano<ref>Template:Cite web</ref>, Neovim<ref>Template:Cite web</ref> and Vim<ref>Template:Cite web</ref>. Programming languages such as Assembly, C, C++, Haskel, Java, Lisp, Node.js<ref>Template:Cite web</ref>, Octave, OCaml<ref>Template:Cite web</ref>, PHP<ref>Template:Cite web</ref>, Perl, Picolisp, Python, Ruby, Rust<ref>Template:Cite web</ref> and TeX<ref>Template:Cite web</ref> are included. And it's just a normal app, not requiring root or anything. 

Addons

Termux has some extra features. You can add them by installing addons: 

Termux:API<ref>Template:Cite web</ref> Access Android, Chromebook and Fire OS hardware features. 

Termux:Boot<ref>Template:Cite web</ref> Run script(s) when your device boots. 

Termux:Float<ref>Template:Cite web</ref> Run Termux in a floating window. 

Termux:Styling<ref>Template:Cite web</ref> Have color schemes and powerline-ready fonts customize the appearance of the Termux terminal. 

Termux:Task<ref>Template:Cite web</ref> An easy way to call Termux executables from Tasker and compatible apps. 

Termux:Widget<ref>Template:Cite web</ref> Start small scriptlets from the home screen. 

References

Template:Reflist 




External links
• FAQ at The Termux Wiki
• IRC: Channel #Termux on freenode that is mirrored with Gitter.
• Github: Source code and project repository at GitHub. Submit bug reports regarding the main app, packages, and addons. Create pull requests for your solutions.
• Gitter: Chatroom that is mirrored with IRC.
• TermuxTemplate:Software-stub

What is you opinion about https://wiki.termux.com/wiki/Wikipedia the multilingual aspect?

**Comment by sdrausty** [comment]
> Revision as of 21:27, 6 December 2017

Template:Article for deletion/dated Template:Refimprove Template:Infobox software 

Termux is an open source terminal emulator and Linux environment app for Android<ref>Template:Cite web</ref><ref>Template:Cite web</ref><ref>Template:Cite web</ref>, Chromebook<ref>Template:Cite web</ref><ref>Template:Cite web</ref> and Fire OS. It works directly with no rooting or setup required. A minimal base system is installed automatically; Additional packages are available using the APT package manager. 

If you install Termux, you get a full Linux experience with BASH<ref>Template:Cite web</ref>, FISH, MOSH, TCSH, ZSH, and even things like Clang<ref>Template:Cite web</ref>, Emacs<ref>Template:Cite web</ref>, GCC, Nano<ref>Template:Cite web</ref>, Neovim<ref>Template:Cite web</ref> and Vim<ref>Template:Cite web</ref>. Programming languages such as Assembly, C, C++, Haskel, Java, Lisp, Node.js<ref>Template:Cite web</ref>, Octave, OCaml<ref>Template:Cite web</ref>, PHP<ref>Template:Cite web</ref>, Perl, Picolisp, Python, Ruby, Rust<ref>Template:Cite web</ref> and TeX<ref>Template:Cite web</ref> are included. And it's just a normal app, not requiring root or anything. 

Addons

Termux has some extra features. You can add them by installing addons: 

Termux:API Access Android, Chromebook and Fire OS hardware features. 

Termux:Boot Run script(s) when your device boots. 

Termux:Float Run Termux in a floating window. 

Termux:Styling Have color schemes and powerline-ready fonts customize the appearance of the Termux terminal. 

Termux:Task An easy way to call Termux executables from Tasker and compatible apps. 

Termux:Widget Start small scriptlets from the home screen. 

References

Template:Reflist 




External links
• FAQ at The Termux Wiki
• IRC: Channel #Termux on freenode that is mirrored with Gitter.
• Github: Source code and project repository at GitHub. Submit bug reports regarding the main app, packages, and addons. Create pull requests for your solutions.
• Gitter: Chatroom that is mirrored with IRC.
• TermuxTemplate:Software-stub


Ref https://wiki.termux.com/index.php?title=WikipediaEN&diff=3554&oldid=3445

**Comment by Quasic** [comment]
My thoughts on the multilingual aspect:
1. Other language Wikipedias do have different inclusion criteria, as well as other policies. It would be nice to have an editor familiar with them, for each language.
2. Translations are not currently implemented in termux, (though that's not necessarily relevant.)
3. Besides language Wikipedias, there are also wikicommons, wikidata, and possibly more projects that could be relevant.

**Comment by sdrausty** [comment]
Here is an interesting results page https://duckduckgo.com/?q=termux+site%3Aen.wikipedia.org&atb=v98-6_g&ia=web without many real results. 


All Regions▼

Safe Search: Off▼

Any Time▼




Showing results from:en.wikipedia.org All Results





Thermax - Wikipedia

Thermax was founded as a family concern in 1966 by A.S. Bhathena and later taken over by his son in law, Rohinton Aga (R.D. Aga ) ...


https://en.wikipedia.org/wiki/Thermax



List of Linux distributions - Wikipedia

List of Linux distributions Timeline of the development of various Linux ... Debian-based Termux is an open source terminal emulator and Linux environment app for ...


https://en.wikipedia.org/wiki/List_of_Linux_distributions



TeX Live - Wikipedia

TeX Live is a free software distribution for the TeX typesetting system that includes major TeX-related programs, macro packages, and fonts. ... Termux and Gentoo.


https://en.wikipedia.org/wiki/TeX_Live



Web Application Messaging Protocol - Wikipedia

Web Application Messaging Protocol. This article contains content that is written like an advertisement. Please help improve it by removing promotional content and ...


https://en.wikipedia.org/wiki/Web_Application_Messaging_Protocol



User:HsfBot - Wikipedia

User:HsfBot. This user account is a bot operated by Hidayatsrf . It is not a sock puppet, but rather an automated or semi-automated account ...


https://en.wikipedia.org/wiki/User:HsfBot



Rust (programming language) - Wikipedia

Rust is a systems programming language sponsored by Mozilla Research, which describes it as a "safe, concurrent, practical language," supporting functional and ...


https://en.wikipedia.org/wiki/Rust-lang



Wikipedia:WikiProject Java/List of article recents - Wikipedia

Mobile Service Architecture; Spring Roo; Freeplane; TopBT; Apache ZooKeeper; JNAerator ? From bot (AlexNewArtBot): (These articles are the ones recently identified as ...


https://en.wikipedia.org/wiki/Wikipedia:WikiProject_Java/List_of_a...



OCaml - Wikipedia

OCaml (/ oʊ ˈ k æ m əl / oh-KAM-əl), originally named Objective Caml, is the main implementation of the programming language Caml, created by Xavier Leroy ...


https://en.wikipedia.org/wiki/Objective_Caml_(programming_language)



Wikipedia:WikiProject Software - Wikipedia

05 Dec 2017 - Termux ... you can keep in touch with project developments by clicking here to add Wikipedia:WikiProject Software to your watchlist.


https://en.wikipedia.org/wiki/Wikipedia:WPSOFTWARE



Wikipedia:WikiProject Software/Article alerts - Wikipedia

05 Dec 2017 - Termux (talk · edit · hist) was AfDed by Codename Lisa (t · c); see discussion (3 participants).


https://en.wikipedia.org/wiki/Wikipedia:WikiProject_Software/Artic...






No more results.

**Comment by Atavic** [comment]
A minimal stub is up now.

**Comment by sdrausty** [comment]
@Atavic Awesome news!  https://en.wikipedia.org/wiki/Termux is back up!  Hooray!  **Let's work together and keep to keep our article online.**

**Comment by sdrausty** [comment]
{{Infobox software
| name = Termux
| logo =
| logo alt =
| logo caption =
| screenshot =
| screenshot alt =
| caption =
| collapsible =
| author = [https://github.com/fornwall Fredrik Fornwall]
| developer = [https://github.com/termux Termux at GitHub]
| released =
| discontinued =
| ver layout =
| latest release version = [https://github.com/termux/termux-app/releases v0.59]
| latest release date =
| latest preview version =
| latest preview date =
| repo = [https://github.com/termux?tab=repositories Termux Repositories]
| status =
| programming language = [[Java (programming language)|Java]]
| operating system = [[Android (operating system)|Android]], [[Chrome_OS |Chromebook]] and [[Fire_OS |Fire OS]]
| platform =
| size =
| language =
| language count =
| language footnote =
| genre = [[Command-line_interface |Command-line interface]]
and [[Terminal_emulator |Terminal emulator]]
| license = [[Open-source_model |Open source]]
| alexa =
| website = [https://termux.com Termux]
| standard =
| AsOf =
}}

'''Termux''' is a [[Terminal_emulator |terminal emulator]] for [[Android_(operating_system) |Android]] that enables a Linux environment on mobile devices. Termux is an open source solution with a minimal [[List_of_GNU_packages#Base_system |GNU base system]] that includes the [[Bash_(Unix_shell)|Bash]] Unix shell, with additional packages available using Debian [[APT_(Debian) |APT]] package manager.

Addons provide extra features like accessing hardware features as SMS messaging or GPS location directly on Termux.

*From https://en.wikipedia.org/wiki/Termux on 20170114.

**Comment by Quasic** [comment]
This article seems stable. Thanks for creating it. It may not count as the same article. (If it were, it would be redeleted.) Remember to keep this in mind: https://en.wikipedia.org/wiki/Wikipedia:Conflict_of_interest

**Comment by sdrausty** [comment]
@Quasic This page has been created numerous times.  I tried to source the link of the page that was most recently deleted and could not find it.  In spring of 2017 it was locked for a while.  Ditto summer 2017.  The page that was deleted https://github.com/termux/termux-packages/issues/1930#issuecomment-351249489 in December was created by at present unknown author who announced its' creation in our G+ Group.  Who created this version?  

>  This article seems stable. 
>>  This software article is a stub. You can help Wikipedia by expanding it.  

Stub articles in many languages are available https://wiki.termux.com/wiki/Wikipedia here.  English article material is in https://github.com/termux/termux-packages/issues/1930#issuecomment-351254094 this thread here and https://wiki.termux.com/wiki/WikipediaEN here.    **Let's do our best to keep https://en.wikipedia.org/wiki/Termux online by working together.**

**Comment by sdrausty** [comment]
>This article may meet Wikipedia's criteria for speedy deletion as a page that was previously deleted via a deletion discussion, is substantially identical to the deleted version, and any changes do not address the reasons for which the material was deleted. See the previous discussion. See CSD G4. .
.
.
This page was last edited by Codename Lisa (contribs | logs) at 11:20, 15 January 2018 (UTC) (0 seconds ago)

*From https://en.wikipedia.org/wiki/Termux*

**Comment by sdrausty** [comment]
I am shocked, "Termux is being deleted from Wikipedia again.  https://en.wikipedia.org/wiki/Termux The  "This article may meet Wikipedia's criteria for speedy deletion as a page that was previously deleted via a deletion discussion, "" message **does not** show up on smartphone.  How about you?  Are you shocked too?

**Comment by sdrausty** [comment]
> message does not show up on smartphone

**Comment by Quasic** [comment]
It was tagged shortly after I said that, so I spoke too soon. It's to be expected, though. There will be a debate, and an admin will close it. I'm not sure how, yet, but this time it's not as clear-cut. It's one some admins may want to avoid closing. :P

**Comment by sdrausty** [comment]
@Quasic  
> It was tagged

With collective effort we can succeed in having this article https://en.wikipedia.org/wiki/Termux online permanently.  Contribute.  The more meaningful promoting freedom contributions from many users, the  better!  

My posts at G+ are not posting regarding this issue.  Deleted from my log of posts since - never posted.

**Comment by Neo-Oli** [comment]
The wikipedia article was deleted a second time now. I [have asked](https://en.wikipedia.org/wiki/User_talk:Elahrairah#Termux) the administrator to move the article to a draft and he moved it to my userspace. The article can now found here: 

https://en.wikipedia.org/wiki/User:Glow_oli/Termux

There we can continue to improve it and most importantly add sources. Once it is ready we can recreate the page again with the new and improved article. 

I ask you not to recreate the article again or berate the wikipedia admins about this issue any more, so that we don't fall on deaf ears once the article is ready.

**Comment by Atavic** [comment]
Last issue before deletion was *Notability*. That means 3rd party sources are needed, so here some:
https://www.xda-developers.com/termux-the-ultimate-linux-terminal-emulator-for-android-xda-spotlight/
https://www.techrepublic.com/article/how-to-get-a-linux-terminal-on-android/

There's a little banner on the google+ Community:
https://plus.google.com/communities/101692629528551299417

Could it be used as screenshot?

**Comment by Quasic** [comment]
Looking better; now that there is a talk page on Wikipedia, it's the best place to continue this discussion: https://en.wikipedia.org/wiki/User%20talk:Glow_oli/Termux

**Comment by sdrausty** [comment]
@Quasic 
> continue this discussion

With millions, possibly tens of millions of users today, what is the rational behind excluding information about Termux from Wikipedia https://en.m.wikipedia.org/wiki/Termux?  At present this article shows no information about Termux.  Compare information content with https://www.facebook.com/termux/ for example.  

>Installs
1,000,000 - 5,000,000 

*From https://play.google.com/store/apps/details?id=com.termux*

On a side note, I used to install my Termux from https://f-droid.org/packages/com.termux/ so possible a few of the counts at the G Play site might be mine.  Why can't Google count?  I can tell the difference between 1 and 5, my Termux can too.

**Comment by sdrausty** [comment]
I approve this https://en.wikipedia.org/wiki/User:Glow_oli/Termux article for publication.  Do you?

**Comment by Atavic** [comment]
@Neo-Oli As [previous post](https://github.com/termux/termux-packages/issues/1930#issuecomment-358055389), more 3rd party sources:

[https://www.amazon.co.uk/gp/search?index=books&linkCode=qs&keywords=9783319679105](Ad-hoc, Mobile, and Wireless Networks: 16th International Conference on Ad Hoc Networks and Wireless, ADHOC-NOW 2017, Messina, Italy, September 20-22, 2017) 

Termux appears at page 139 under [User-Space Network Tunneling Under a Mobile Platform: A Case Study for Android Environments](http://www.springer.com/gp/book/9783319679099?wt_mc=ThirdParty.SpringerLink.3.EPR653.About_eBook) chapter.

**Comment by Atavic** [comment]
@Neo-Oli more 3rd party sources:

[Ad-hoc, Mobile, and Wireless Networks: 16th International Conference on Ad Hoc Networks and Wireless, ADHOC-NOW 2017, Messina, Italy, September 20-22, 2017](https://www.amazon.co.uk/gp/search?index=books&linkCode=qs&keywords=9783319679105)

*Termux* appears at page 139 under [User-Space Network Tunneling Under a Mobile Platform: A Case Study for Android Environments](http://www.springer.com/gp/book/9783319679099?wt_mc=ThirdParty.SpringerLink.3.EPR653.About_eBook) chapter.

**Comment by sdrausty** [comment]
@Atavic Thank you for the sources.  There are more sources at https://www.facebook.com/termux/  I am trying to add a new newsworthy article per day.  This task is not easy since I do not have my personal web crawler.  After decades of using search engines, I have a firm opinion.  Public search engines hide knowledge in information overload.  To much information = none.  This equals wot (waist of time). 

Our Wikipedia page is high jacked!  It has been high jacked and is presenting zero information about Termux for a long time, year N?  No information = No installation of Termux.

**Our `Learn More` button at https://www.facebook.com/termux/ is high jacked as well!**  Click or tap the `Learn More` button more to find out why.  It is worse than zero information about Termux at present.  Your assistance is requested.  

Editing and deleting this anti-promotion button at Facebook is impossible at present.  No one can edit or  change the links output afaik.

**Comment by sdrausty** [comment]
Our `Learn More` button at https://www.facebook.com/termux/ is high jacked along with the Wikipedia article including Termux references as well since *a little while* ago!  It seems that both Facebook and Wikipedia promote high jackings. Do u agree?

**"Termux is High Jacked"** Front page newspaper headline which *might fulfill* Wikipedia's request for notoriety. 

Does anyone know about the `Learn More` button high jacking at https://www.facebook.com/termux ?

Take a screenshot and save and publish it  - of the empty Wikipedia page and of the high jacked Facebook `Learn More` button. 

"Learn less, not more" This is the creed of Facebook and Wikipedia afaict, in addition to information overload. 

```
creed 

n.  A formal statement of religious belief; a confession of faith.

n.  A system of belief, principles, or opinions...
```

Save and publish it. Every time you check up on it. 

They will say, **"We never high jacked anything. Prove it!"**

Save and publish it. Every time you check up on it. Automate this publishing. It should be simple:
1) Check the link.
2) Make your screenshot automated with published high jacked Termux throughout the Internet in one tap. 

**Wikipedia year N! Now Facebook week N!** 

**Termux is high jacked from the public!** It took me months of trolling and selecting to find Termux and I love it. I love the freedom from desktop/similar Termux brings.

Are there any desktop/laptop/notebook manufacturer representatives here? Do these people want everyone to be stuck in a made of plastic windowless sans Windows(TM) cubical with a clunky desktop machine with peripherals? 

The Termux `Learn More` button at https://www.facebook.com/termux/ is high jacked along with the Termux Wikipedia article and its' references. Do you know about these very unpleasant facts regarding Termux?

**Comment by sdrausty** [comment]
> blocked and blocking knowledge Wikipedia article 

Wikipedia keeps us all ignorant as long as we believe in its' written word.

**Comment by Atavic** [comment]
@sdrausty I have no Facebook.

**Comment by Quasic** [comment]
I have no Facebook, either.

I don't see any difference in the article, but we can look at the links and incorporate them.

Wikipedia shouldn't be trusted. There are too many conflicts and vandals. It can be useful to find more helpful resources, though.

**Comment by Quasic** [comment]
Please don't recreate the article. It only hurts. If an article is recreated too often, it is salted (blacklisted) which then requires an admin to create it.

**Comment by Quasic** [comment]
@Atavic, about the Google+ banner, we would need to upload a copy, either with verified permission of the creator or with a claim that it is too simple to worry about copyright, as it's basically just $ Termux []

**Comment by sdrausty** [comment]
> Please don't recreate the article. It only hurts. If an article is recreated too often, it is salted (blacklisted) 

Who asked me in early summer, "@sdrausty, why don't you write it?".

**Comment by Grimler91** [comment]
> > Please don't recreate the article. It only hurts. If an article is recreated too often, it is salted (blacklisted)
> 
> Who asked me in early summer, "@sdrausty, why don't you write it?".

I suggested that you should write it @sdrausty. @Quasic has a point, we can't force wikipedia moderators to accept the article. 
What we can do is to address their concerns, thereby improving the article and then upload it again.

**Comment by Atavic** [comment]
@Quasic

>verified permission 

Eh... I can't find any source for that, BTW:

The image on the right, [here](https://wiki.termux.com/wiki/Main_Page) is hosted on mediawiki, so WP should be happy.

**Comment by Atavic** [comment]
@Grimler

Actually the page is handled by Neo-Oli, see: https://github.com/termux/termux-packages/issues/1930#issuecomment-357994565

**Comment by Atavic** [comment]
@Grimler91 Actually the page is handled by Neo-Oli, see: https://github.com/termux/termux-packages/issues/1930#issuecomment-357994565

**Comment by sdrausty** [comment]
@Gimler
> you should write it

Benny there; done that.

**Comment by gimler** [comment]
@sdrausty i think you mean @Grimler91

**Comment by sdrausty** [comment]
Correct.

**Comment by Grimler91** [comment]
@Atavic yeah, but everyone can contribute to it of course

**Comment by Quasic** [comment]
@Atavic, I don't know the license for the Google+ image, but the wiki one looks compatible. Of course, anyone can take their own screenshot and upload it.

**Comment by kothiha931** [comment]
Hello-World

**Comment by Avi-Sayen** [comment]
Hello, so I created a new one.

**Comment by Avi-Sayen** [comment]
Hello, so I created a new wiki page for termux
[https://en.m.wikipedia.org/wiki/Termux#Features](https://youtu.be/dQw4w9WgXcQ)
Currently the page has only few information and I'm not very good at writing so edits will be appreciated.
